### PR TITLE
[RESTEASY-2851]Fix Java 2 Security Permission in SseEventSourceScheduler.shutdownNow()

### PR DESCRIPTION
**Pull request by jim-krueger**

### CODE REVIEW

**Comment by asoldano** [comment]
@jim-krueger , we require all contributions to be made under the terms of the ASL 2.0 License: http://www.apache.org/licenses/LICENSE-2.0 
Do you agree to these terms?

**Review by asoldano — APPROVED** [review]

**Comment by ronsigal** [comment]
@jim-krueger, do you agree to the terms of the ASL 2.0 License: http://www.apache.org/licenses/LICENSE-2.0?

**Comment by jim-krueger** [comment]
@ronsigal Wow, missed seeing this for a while sorry.  Yes, I agree to the terms of the ASL 2.0 License.

File: resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceScheduler.java [hunk 1]
```diff
@@ -142,7 +145,13 @@ void shutdownNow()
...
          if (!scheduledExecutorService.isManaged())
          {
-            this.scheduledExecutorService.shutdownNow();
+            AccessController.doPrivileged(new PrivilegedAction<Void>() {
+              @Override
+              public Void run() {
+                scheduledExecutorService.shutdownNow();
+                return null;
+              }
+           });
```
Comment (jamezp) [inline resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceScheduler.java:154]: For performance sake we should do something like:
```java
if (System.getSecurityManager() == null) {
    scheduledExecutorService.shutdownNow();
} else {
    AccessController.doPrivileged(new new PrivilegedAction<Void>() {
        @Override
        public Void run() {
            scheduledExecutorService.shutdownNow();
            return null;
        }
    });
}
```

**Review by jamezp — CHANGES_REQUESTED** [review]

**Comment by jamezp** [comment]
Replaced by #3273 3273

**Pull request closed without merging**